dev-manuals/services/overworld/add-keybindings.md

```diff
@@ -0,0 +1,38 @@
+# Add keybindings
+
+## Overview
+
+Each player has their own prefered binding of available actions to keys on the keyboard. Therefor, we support persistent, player specific keybindings across all courses. That means that a player has to configure his bindings only once and can then use them in all his courses.
+
+## How to add a keybinding
+
+If you want to add another action, you have to add it to the list of existing ones, in order to use it and for players to be able to bind it to their prefered key.
+
+### Overworld
+
+You have to do the following in the overworld:
+
+1. You need to add the new action to the script `Binding` located at `Assets/Scripts/Keybinding`.  
+![Bindings enum](assets/keybindings-bindings-enum.webp)
+2. You need to assign a default key in the function `ResetKeybindings` in the script `GameManager` located at `Assets/Scripts/GameManager`. Simply create a `Keybinding` object with your binding and your chosen key as a `KeyCode` object and call the function `ChangeKeybind` with that `Keybinding` object. For reference, take a look at the other keybindings.  
+![Default binding](assets/keybindings-default-binding.webp)
+
+You need to assign an unused key, so you cannot chose a key already binded to another action!
+
+### Overworld Backend
+
+You have to do the following in the overworld backend:
+
+1. You need to add the new action to the enum `Binding` located at `src\main\java\de\unistuttgart\overworldbackend\data\enums`.
+
+It has to be the same action as in the overworld!
```

Comment:
:frowning_face: And yet another thing I have to fix.
That's not the task of the backend.
That's basically as bad as storing map coordinates hardcoded in the backend.
The backend should simply store some keybindings data.
The frontend queries this data and assigns meaning to it.
The backend should be nothing else than a persistent storage in this case.

Reply:
Oh, also
```suggestion
1. You need to add the new action to the enum `Binding` located at `src/main/java/de/unistuttgart/overworldbackend/data/enums/`.

It has to be the same action as in the overworld!
```
Additionally, even if this enum miraculously prevails, it should be renamed to `KeyBinding` or something similar.